### PR TITLE
[Driver][SYCL] Add support for exporting symbols for AOT

### PR DESCRIPTION
**Pull request by mdtoguchi**
Add support for -ftarget-export-symbols.  This option when used with -fsycl -fsycl-targets=spir64_gen when creating shared libraries.  This exposes the exported symbols in the library so they can be available for other modules.

### CODE REVIEW

File: sycl/doc/UsersManual.md [hunk 1]
```diff
@@ -410,6 +410,13 @@ and not recommended to use in production environment.
     Also, some of Intel GPUs or GPU run-time/drivers may support only
     "stateless" memory accesses.
 
+**`-f[no-]target-export-symbols`**
+
+    Exposes exported symbols in a generated target library to allow for
+    visibility to other modules.
+
+    NOTE: This flag is only supported for spir64_gen AOT targets.
```
Comment (srividya-sundaram) [inline sycl/doc/UsersManual.md:418]: What is the expected behavior if this flag is used for targets other than `spir64_gen` ?

Comment (mdtoguchi) [reply sycl/doc/UsersManual.md:418]: The option will effectively be a no-op.  Under non AOT spir64_gen conditions, the driver will emit a warning that the option is unused.

Comment (srividya-sundaram) [reply sycl/doc/UsersManual.md:418]: Maybe we could consider adding a test case to capture this case.

**Comment by mdtoguchi** [comment]
@intel/dpcpp-clang-driver-reviewers, @intel/dpcpp-doc-reviewers, please take a look.  Thanks!

**Review by AlexeySachkov — APPROVED** [review]

**Review by hchilama — APPROVED** [review]
LGTM

**Comment by mdtoguchi** [comment]
@intel/llvm-gatekeepers, this one should be ready to go!  thanks

**Pull request merged**